### ISSUE TITLE
CRuby master reports 0.2.0 but seems newer

### ISSUE THREAD

**Issue by eregon** [ruby/ostruct#20]
```
ruby -v -rostruct -e 'p OpenStruct::VERSION'
ruby 3.0.0dev (2020-10-23T06:26:51Z master f754b42285) [x86_64-linux]
"0.2.0"
```

Yet the version of OpenStruct in CRuby master seems more recent than 0.2.0, notably it uses the new variant for `YAML.dump`.

I guess CRuby master is missing some fixes from this repo?
I think CRuby master should use a release of OpenStruct (probably the latest), and not something in between.
cc @marcandre

**Comment by marcandre** [comment]
Sorry, haven't had a chance to look at this yet...

**Comment by eregon** [comment]
Am I right that the changes in the YAML dumping are from OpenStruct 0.3, and not in OpenStruct 0.2?
CRuby master uses the new format, yet reports as OpenStruct 0.2 when I filed this issue.
So the spec written by @headius failed and I had to allow both formats without any check in https://github.com/ruby/spec/commit/4f59d86898fdd397db077209cfc50dc0f784fcf4.
That's not a big deal on its own, just surprising.

Which is one more reason for https://bugs.ruby-lang.org/issues/16778 of course, then the versions would be true and not "some vague approximation" because ruby/ruby master reports as some old version + cherry-picked changes.
